### PR TITLE
Feature/doc component

### PR DESCRIPTION
**Pull request by JochenKienzle**

### CODE REVIEW

**Comment by JochenKienzle** [comment]
For better read-ability of iotea-component.md please have a look to here: [iota-components](https://github.com/boschglobal/iot-event-analytics/blob/feature/doc_component/docs/topics/iotea-components.md)

File: docs/topics/assets/plantuml/components_platform.puml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    node Platform #LightBlue {
+        package Pipeline #Ivory {
+            component Ingestion
+            component Encoding
+            component Routing
+        }
+
+        note bottom of Pipeline
+            Multiple instancesP
```
Comment (mariaivanova-git) [inline docs/topics/assets/plantuml/components_platform.puml]: instancesP -> instances

File: docs/topics/iotea-components.md [hunk 2]
```diff
@@ -8,68 +8,132 @@
...
-- Sends rules updates to all Rule-Managers
-- Holds Metadata-Manager master instance which is the single master loading metadata for static segments/types and features
-- Sets indices for incoming dynamic features from talents
-- Updates all Metadata-Manager worker instances via MQTT
-- __Rules-Manager__
-  - Holds internal rule state (for all talents)
-  - Synchronized via MQTT by Config-Manager
+- [Platform Components](#Platform-Components)
+- [SDK Components](#SDK-Components)
+- [Utils](#Utils-Components)
```
Comment (mariaivanova-git) [inline docs/topics/iotea-components.md]: No Utils or Plugins Section, so links do not work.

File: docs/topics/iotea-components.md [hunk 3]
```diff
@@ -8,68 +8,132 @@
...
+
+- Mapper
+- Worker
+- Reducer
+
+# VSCode Extension
+
+The VSCode extension provides setup routines, auto-completion and other helpers to develop, test and run [_Talents_](#Talent)).
+
+Installation instructions and further information can be found [here](../../../src/sdk/vscode/README.md).
```
Comment (mariaivanova-git) [inline docs/topics/iotea-components.md]: link  should be ../../src/sdk/vscode/README.md

File: docs/topics/iotea-components.md [hunk 4]
```diff
@@ -8,68 +8,132 @@
...
+## ConfigManager
 
-- Encodes incoming events by various encoders (minmax, delta, categorical) --> Not implemented yet
-- Uses Metadata-Manager Slave to retrieve metadata
-- Uses Instance-Manager to store an encoded field into the appropriate instance
-- Freely scalable by ingestion instance count
+The ConfigManager manages and provides the platform configuration. This is done statically via a configuration file (_config.json_).
 
-### Routing
+The ConfigManager provides a periodic [_Talent_](#Talent) discovery and provides platform configuration (e.g. available _Types_ and _Features_) and dynamically via a periodic _Talent_ discovery. It holds the [_MetadataManager_](#MetadataManager) and [TalentConfigManager](#TalentConfigManager) master instances and synchronize all other instances of them in the system.
```
Comment (mariaivanova-git) [inline docs/topics/iotea-components.md]: synchronize -->  synchronizes

File: docs/topics/iotea-components.md [hunk 5]
```diff
@@ -8,68 +8,132 @@
...
 
-- Freely scalable by ingestion instance count
-- Uses rules to determine whether an event should be forwareded to a specific talent
-  - Rules can be any type of boolean algebra (containing and/or rules)
-  - Enriches event with all fields a talent is "interested" in
+## Ingestion
 
-### Talent
+This component is used to transform and validate incoming events and it's configurable as an ETL-pipeline and Multi-Channel supports an arbitrary amount of parallel ETL-pipelines and its completely configurable via configuration files (json) and scalable via multiple instances.
```
Comment (mariaivanova-git) [inline docs/topics/iotea-components.md]: A bit unclear, especially in the part of Multi-Channel. Maybe some splitting into sentences?

its -> it's

File: docs/topics/iotea-components.md [hunk 6]
```diff
@@ -8,68 +8,132 @@
...
+
+This component encodes incoming events by various encoders (minmax, delta, categorical) and it uses the InstanceManager to store an encoded field into the appropriate instance. It's scalable via multiple instances.
+
+## Routing
+
+This component routes the events to specific [_Talents_](#Talent) by evaluating their related rule-set. It's scalable via multiple instances.
+
+## Sub Components
+
+This components are used by the other platform components and act like helper or has been extracted to make them re-usable.
```
Comment (mariaivanova-git) [inline docs/topics/iotea-components.md]: This components --> These components

has been extracted --> have been extracted

File: docs/topics/iotea-components.md [hunk 7]
```diff
@@ -8,68 +8,132 @@
...
+
+- isSet, change, greater than, less than, ...
+- time series constraints
+- Nelson constraints
+
+### ProtocolGateway
+
+The ProtocolGateway abstracts the publish-subscribe protocol from a specific implementation of it. Therefore, a _ProtocolGatewayAdapter_ is used to integrate the underlying publish-subscribe communication (e.g. MQTT Broker).
+
+This abstraction is not limited for platform components. It is also the common way to abstract the clients ([_Talents_](#Talent)).
```
Comment (mariaivanova-git) [inline docs/topics/iotea-components.md]: to abstract the clients (Talents)
  suggestion -->
to abstract the clients (Talents) communication

File: docs/topics/iotea-components.md [hunk 8]
```diff
@@ -8,68 +8,132 @@
...
+The SDK components are used to implement clients (which are called [_Talents_](#Talent)) for IoT Event Analytics.
+
+SDKs are available for following languages:
+
+- NodeJS
+- Python
+- C++
+- (Experimental) NodeRed
+
+**Note**: Not all SDKs shares the same component structure, names and additional features.
```
Comment (mariaivanova-git) [inline docs/topics/iotea-components.md]: shares --> share

**Review by mariaivanova-git — CHANGES_REQUESTED** [review]
Added a few minor editorial comments.

File: docs/topics/iotea-components.md (hunk 5 above)
Comment (JochenKienzle) [reply docs/topics/iotea-components.md]: Done

File: docs/topics/iotea-components.md (hunk 6 above)
Comment (JochenKienzle) [reply docs/topics/iotea-components.md]: Done

File: docs/topics/iotea-components.md (hunk 7 above)
Comment (JochenKienzle) [reply docs/topics/iotea-components.md]: Done

File: docs/topics/iotea-components.md (hunk 8 above)
Comment (JochenKienzle) [reply docs/topics/iotea-components.md]: Done

File: docs/topics/assets/plantuml/components_platform.puml (hunk 1 above)
Comment (JochenKienzle) [reply docs/topics/assets/plantuml/components_platform.puml]: Done

File: docs/topics/iotea-components.md (hunk 2 above)
Comment (JochenKienzle) [reply docs/topics/iotea-components.md]: I have removed utils and have added an Plugins/Extension section

File: docs/topics/iotea-components.md (hunk 3 above)
Comment (JochenKienzle) [reply docs/topics/iotea-components.md]: Done

File: docs/topics/iotea-components.md (hunk 4 above)
Comment (JochenKienzle) [reply docs/topics/iotea-components.md]: Done

File: docs/topics/iotea-components.md (hunk 2 above)
Comment (mariaivanova-git) [reply docs/topics/iotea-components.md]: The [Plugins / Extensions](##Plugins-/-Extensions) link seems not to work.

**Comment by llakie** [comment]
Please rebase onto current develop to avoid failing integration tests

**Comment by JochenKienzle** [comment]
--> Rebased.
Please check

**Review by llakie — APPROVED** [review]

**Comment by JochenKienzle** [comment]
@mariaivanova-git Are there any further review findings or is it ok from your side?
@llakie After this has been clarified I think you can merge it 👍

**Comment by mariaivanova-git** [comment]
@JochenKienzle Only this comment "The Plugins / Extensions link seems not to work." seems not resolved. No new findings.

**Review by mariaivanova-git — APPROVED** [review]

**Pull request merged**